### PR TITLE
Add output variable support

### PR DESCRIPTION
**Pull request by raghavharness**

### CODE REVIEW

File: pkg/output/output.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package output
+
+import (
+	"github.com/joho/godotenv"
+)
+
+func WritePluginOutputFile(outputFilePath, imageName, digest string, tags []string) error {
```
[the remark this answers is not in the document]
Comment (raghavharness) [reply pkg/output/output.go]: fixed, thanks

**Review by shubham149 — APPROVED** [review]

File: kaniko.go [hunk 2]
```diff
@@ -223,19 +230,29 @@ func (p Plugin) Exec() error {
...
+		}
+	}
+
 	return nil
 }
 
+func getDigest(digestFile string) string {
+	content, err := ioutil.ReadFile(digestFile)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to read digest file contents at path: %s with error: %s\n", digestFile, err)
```
Comment (eoinmcafee00) [inline kaniko.go:251]: Check for errors when calling getDigest: It's possible that the call to getDigest could fail and return an error.

File: kaniko.go [hunk 3]
```diff
@@ -223,19 +230,29 @@ func (p Plugin) Exec() error {
...
-		}
-		err = artifact.WritePluginArtifactFile(p.Artifact.RegistryType, p.Artifact.ArtifactFile, p.Artifact.Registry, p.Artifact.Repo, string(content), p.Artifact.Tags)
+		err = artifact.WritePluginArtifactFile(p.Artifact.RegistryType, p.Artifact.ArtifactFile, p.Artifact.Registry, p.Artifact.Repo, getDigest(p.Build.DigestFile), p.Artifact.Tags)
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "failed to write plugin artifact file at path: %s with error: %s\n", p.Artifact.ArtifactFile, err)
 		}
 	}
 
+	if p.Output.OutputFile != "" {
+		if err = output.WritePluginOutputFile(p.Output.OutputFile, getDigest(p.Build.DigestFile)); err != nil {
```
Comment (eoinmcafee00) [inline kaniko.go:240]: Handle any errors when writing the output file: If there is an error when writing the plugin output file, this code only logs the error message and continues executing.

Comment (raghavharness) [reply kaniko.go:240]: Yeah that is the behaviour we want currently. This is done just as a starter to get the product view on this. Once we get the feedback we'll improve this.

**Review by eoinmcafee00 — APPROVED** [review]

**Pull request merged**